docs/source-pytorch/starter/lightning_lite.rst

```diff
@@ -725,3 +725,29 @@ the data is written to disk.
     self.barrier()
 
     # All processes are allowed to read the data now
+
+
+no_backward_sync
+================
+
+Use this context manager when performing gradient accumulation and using a distributed strategy (e.g., DDP).
+It will speed up your training loop by cutting redundant communication between processes during the accumulation phase.
+
+.. code-block:: python
+
+    # Accumulate gradient 8 batches at a time
+    is_accumulating = batch_idx % 8 != 0
+
+    with self.no_backward_sync(model, enabled=is_accumulating):
+        output = model(input)
+        loss = ...
+        self.backward(loss)
+        ...
+
+    # Step the optimizer every 8 batches
+    if not is_accumulating:
+        optimizer.step()
+        optimizer.zero_grad()
+
+Not all strategies support it. Both the model's `.forward()` and the `self.backward()` call need to run under
```

Comment:
Can you list which strategies don't support it ? I think DeepSpeed might work with this one no ?

Reply:
I can, the list will probably get outdated but we will try to keep it updated. 
My understanding is that [DeepSpeed handles accumulation internally](https://ghpages-test.readthedocs.io/en/latest/config.html#deepspeed.config.BatchConfig.gradient_accumulation_steps) in the DeepSpeedEngine (you configure it through the config dict/file), and thus we don't have influence on it.